src/hotspot/os/linux/cgroupV1Subsystem_linux.hpp

```diff
@@ -51,18 +50,32 @@ class CgroupV1Controller: public CgroupController {
     char *subsystem_path() { return _path; }
 };
 
-class CgroupV1MemoryController: public CgroupV1Controller {
+class CgroupV1MemoryController: public CgroupV1Controller, public CgroupMemoryController {
 
   public:
     bool is_hierarchical() { return _uses_mem_hierarchy; }
     void set_subsystem_path(char *cgroup_path);
+    jlong read_memory_limit_in_bytes(julong upper_bound);
+    jlong memory_usage_in_bytes();
+    jlong memory_and_swap_limit_in_bytes(julong host_mem, julong host_swap);
+    jlong memory_and_swap_usage_in_bytes(julong host_mem, julong host_swap);
+    jlong memory_soft_limit_in_bytes(julong upper_bound);
+    jlong memory_max_usage_in_bytes();
+    jlong rss_usage_in_bytes();
+    jlong cache_usage_in_bytes();
+    jlong kernel_memory_usage_in_bytes();
+    jlong kernel_memory_limit_in_bytes(julong host_mem);
+    jlong kernel_memory_max_usage_in_bytes();
+    void print_version_specific_info(outputStream* st, julong host_mem);
```

Comment:
Can you add `override` to the end of the virtual functions here? `jlong memory_usage_in_bytes() override;`, like so.